### ISSUE TITLE
Modifying session from spark?

### ISSUE THREAD

**Issue by remy** [zemirco/primus-express-session#3]
I can't work out if it's possible for the socket to update the session. My early tests are failing and nothing's being saved.

I've created a gist based on your express 4.x code.

This line should read the session: https://gist.github.com/remy/ad3c853b070b7a1490a8#file-app-js-L42

And this is where I'm setting and saving the session: https://gist.github.com/remy/ad3c853b070b7a1490a8#file-app-js-L73

The idea being that the socket opens, and data from the socket is saved in a session, so when they revisit (or refresh in this case) the session state is still there.

Is this possible?

**Comment by remy** [comment]
I've been investigating further and have worked out that the session is somehow reset when using the `browserchannel` transporter. I used the `memoryStore` in expression for the session, and added some logging around [this method](https://github.com/expressjs/session/blob/master/session/memory.js#L81), and noticed an extra (poll?) request coming in via browserchannel where the session was in a previous state (I should add this doesn't make sense, because the session is looked up based on the `sid` in `memoryStore` and so it should be consistent either way...but apparently not).

**Comment by zemirco** [comment]
So far I've always initiated the session from Express. Not the other way around. Will try to work something out. Have you seen the [middleware example](https://github.com/primus/primus/tree/master/examples/middleware)? It also sets the session from Express but maybe it can help.

In the meantime, maybe @3rd-Eden has an idea?!

**Comment by 3rd-Eden** [comment]
I don't see any obvious failures in the code you posted @remy, i'll see if I can reproduce it locally and figure out what its going on here.

**Comment by 3rd-Eden** [comment]
@remy I cannot reproduce your issue. I've taken the index.js file that you've posted and and modified your `/` route to just serve a basic HTML file with the contents of:

``` js
<script src="/primus/primus.js"></script>
<script>
  primus = new Primus();
</script>
```

And when I run the code i get: 

```
body-parser deprecated undefined extended: provide extended option index.js:24:20
express-session deprecated undefined resave option; provide resave option index.js:27:15
express-session deprecated undefined saveUninitialized option; provide saveUninitialized option index.js:27:15
express-session deprecated req.secret; provide secret option index.js:27:15
Express server listening on port 3000
on::connection
john
/ undefined                                           ----- FIRST REFRESH, undefined as expected
GET / 200 10.137 ms - -
on::connection
john                                                      ----- output of the req.session.username from primus
/ john                                                    ----- SECOND REFRESH HERE output from the request handler
GET / 200 5.263 ms - -
on::connection
john
```

**Comment by remy** [comment]
Yeah, in retrospect, I'm pretty damn certain it's because of the `browserchannel` transporter. @3rd-Eden if you try with that transporter, and not `websockets` do you get the same result?

**Comment by 3rd-Eden** [comment]
@remy Yes, your right. It seems to be the transport that messes this up. The second refresh still shows undefined, but the third did show `john`. I'll see if I can find something useful in the source that would lead to this behavior
